### ISSUE TITLE
S4670 false positive with angular material

### ISSUE THREAD

**Issue by lnu** [SonarSource/sonar-css#151]
We use [angular material](https://github.com/angular/material2) and sonar css analyzer reports bugs like this one:
![image](https://user-images.githubusercontent.com/1829553/52571102-36cb6100-2e15-11e9-8d6a-bc9993849c8f.png)

There is a paremeter in StyleLint for that situation:
```
"rules": {
     "selector-type-no-unknown": [true, { "ignoreTypes": ["/^mat-/","/^retrace-/"] }]
  }
```

Would it be possible to have a parameter like this in sonar?

**Comment by MaxTranced** [comment]
This would be really useful to fix, IMO this should be more than just an "enhancement".

**Comment by vilchik-elena** [comment]
To fix this problem we should probably (re)use stylelint parameter.

**Comment by ijavid** [comment]
`.stylelintrc` file should be respected by SonarCSS

**Comment by vilchik-elena** [comment]
@lnu You set `"/^retrace-/"` to the ignored types, can you give a link to the docs? I can't find anything similar

**Comment by SaiVikas0801** [comment]
Hi,

I am also facing a similar issue. My project is based out of ionic & angular and when sonar report is being analysed its shows up similar. It is not able to identify all the ioniic tags.
Please find the attached screenshot for reference. 

![sonar_error](https://user-images.githubusercontent.com/28046714/64855370-5b9ee580-d63d-11e9-820c-e3af45b2e92b.png)

Can you please suggest how to resolve this issue.

**Comment by andreicristianpetcu** [comment]
@SaiVikas0801 Material components usually add a class with the same name. I would try .ion-footer and .ion-icon.

**Comment by SaiVikas0801** [comment]
> @SaiVikas0801 Material components usually add a class with the same name. I would try .ion-footer and .ion-icon.

Hi andreipetcu, 

So you want me to try changing the syntax in scss file by referring the element with class name ?

**Comment by songokudbz** [comment]
How do you do this for regular css files?

**Comment by thw0rted** [comment]
I'd like to direct anybody watching this issue to [this Community post I just made](https://community.sonarsource.com/t/stop-flagging-valid-custom-web-component-tag-names-with-s4670/30262) which is relevant to your interests.  Short version: stop checking tag names with dashes in them.

**Comment by josegironp** [comment]
Hi, I have the same problem, does anyone know how I can fix this?
